posthog/temporal/delete_recordings/workflows.py

```diff
@@ -0,0 +1,86 @@
+import json
+import asyncio
+from datetime import timedelta
+
+from temporalio import common, workflow
+from temporalio.workflow import ParentClosePolicy
+
+from posthog.temporal.common.base import PostHogWorkflow
+from posthog.temporal.delete_recordings.activities import (
+    delete_recording_blocks,
+    load_recording_blocks,
+    load_recordings_with_person,
+)
+from posthog.temporal.delete_recordings.types import (
+    DeleteRecordingBlocksInput,
+    DeleteRecordingInput,
+    DeleteRecordingsWithPersonInput,
+    LoadRecordingBlocksInput,
+    LoadRecordingsWithPersonInput,
+)
+
+
+@workflow.defn(name="delete-recording")
+class DeleteRecordingWorkflow(PostHogWorkflow):
+    @staticmethod
+    def parse_inputs(input: list[str]) -> DeleteRecordingInput:
+        """Parse input from the management command CLI."""
+        loaded = json.loads(input[0])
+        return DeleteRecordingInput(**loaded)
+
+    @workflow.run
+    async def run(self, input: DeleteRecordingInput) -> None:
+        recording_blocks = await workflow.execute_activity(
+            load_recording_blocks,
+            LoadRecordingBlocksInput(session_id=input.session_id, team_id=input.team_id),
+            start_to_close_timeout=timedelta(minutes=1),
+            retry_policy=common.RetryPolicy(
+                maximum_attempts=2,
+                initial_interval=timedelta(minutes=1),
+            ),
+            heartbeat_timeout=timedelta(seconds=10),
+        )
+
+        await workflow.execute_activity(
+            delete_recording_blocks,
+            DeleteRecordingBlocksInput(session_id=input.session_id, team_id=input.team_id, blocks=recording_blocks),
+            start_to_close_timeout=timedelta(minutes=1),
+            retry_policy=common.RetryPolicy(
+                maximum_attempts=2,
+                initial_interval=timedelta(minutes=1),
+            ),
```

Comment:
i wonder if we should allow more retries

or rather, i wonder (like prom counters above) how we would know if these were failing once retry exhausted